### ISSUE TITLE
2D linear upsample with `align_corners=False` doesn't match PyTorch.

### ISSUE THREAD

**Issue by ysiraichi** [pytorch/xla#8897]
## 🐛 Bug

Running a 2D linear upsample yields in different results when comparing with PyTorch (both CPU and CUDA). That, however, only happens when `align_corners=False`.

```python
def run(device):
    a = torch.arange(9, dtype=torch.float32, device=device).view(1, 1, 3, 3)
    m = Upsample(a, scale_factor=1.5, mode="bilinear", align_corners=False)
    return m(a)

>>> run("cpu")
tensor([[[[0.0000, 0.5000, 1.1667, 1.8333],
          [1.5000, 2.0000, 2.6667, 3.3333],
          [3.5000, 4.0000, 4.6667, 5.3333],
          [5.5000, 6.0000, 6.6667, 7.3333]]]],
       grad_fn=<UpsampleBilinear2DBackward0>)

>>> run("xla")
tensor([[[[0.0000, 0.6250, 1.3750, 2.0000],
          [1.8750, 2.5000, 3.2500, 3.8750],
          [4.1250, 4.7500, 5.5000, 6.1250],
          [6.0000, 6.6250, 7.3750, 8.0000]]]], device='xla:0',
       grad_fn=<UpsampleBilinear2DBackward0>)
```

PyTorch/XLA result seems to align with tensorflow (makes sense, since we took it from there), and OpenCV. However, PyTorch does seem to be the different one here.

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]: CUDA
 - torch_xla version: 3808df0162b2704f9e1913a066a97f49955aedde

## Additional context

Found this when lowering the backwards (issue #8754). I tried taking the lowering from [the same place](https://github.com/tensorflow/tensorflow/blob/e51d6ab5730092775d516b18fa4ee85d49602cd8/tensorflow/compiler/tf2xla/kernels/image_resize_ops.cc#L731-L808) and it looks like the gradient also matches only when `align_corners=True`.

This might not affect XLA:TPU, since it uses a custom lowering.

**Comment by ysiraichi** [comment]
**Question:** are all of these correct results of a different interpretation?

**Comment by ysiraichi** [comment]
Apparently, this mismatch is due to PyTorch using the `scale_factor`, while tensorflow (and OpenCV) uses the ratio between output and input sizes. If we call `Upsample(..., recompute_scale_factor=True)` we get the same result.

**Comment by ysiraichi** [comment]
I'm opening this issue, since with `recompute_scale_factor=False`, we still get different results.
